test/e2e/ProvisionE2ETest.php

```diff
@@ -0,0 +1,29 @@
+<?php
+namespace e2e;
+
+use Deployer\Exception\Exception;
+
+class ProvisionE2ETest extends AbstractE2ETest
+{
+    private const RECIPE = __DIR__ . '/recipe/provision.php';
+
+    /**
+     * @throws Exception
+     */
+    public function testProvisionRecipe(): void
+    {
+        $this->init(self::RECIPE);
+
+        $this->tester->setInputs(['deployer']); // set input for deployer user pass
+
+        $this->tester->run([
+            'provision',
+            '-f' => self::RECIPE,
+            '-o' => [ 'firewall=false' ], // disable firewall provisioning in docker e2e environment
+            'selector' => 'all'
+        ]);
+
+        $display = trim($this->tester->getDisplay());
+        self::assertEquals(0, $this->tester->getStatusCode(), $display);
```

Comment:
Maybe also run some command to check what provision did succeed? Like check php version?

Reply:
Ok, will add tests that verify presence of PHP, PHP-FPM & nginx